src/flirtlib.cpp

```diff
@@ -39,52 +39,87 @@
 #include <flirtlib_ros/flirtlib.h>
 #include <flirtlib_ros/conversions.h>
 #include <boost/foreach.hpp>
+#include <boost/iterator/zip_iterator.hpp>
 
 namespace flirtlib_ros
 {
 
-using std::string;
-namespace sm=sensor_msgs;
-
-Detector* createDetector (SimpleMinMaxPeakFinder* peak_finder)
-{
-  const double scale = 5.0;
-  const double dmst = 2.0;
-  const double base_sigma = 0.2;
-  const double sigma_step = 1.4;
-  CurvatureDetector* det = new CurvatureDetector(peak_finder, scale, base_sigma,
-                                                 sigma_step, dmst);
-  det->setUseMaxRange(false);
-  return det;
-}
-
-DescriptorGenerator* createDescriptor (HistogramDistance<double>* dist)
+FlirtlibFeatures::FlirtlibFeatures (ros::NodeHandle nh)
+  : nh_(nh)
+  , detector_type_("curvature")
+  , descriptor_type_("Shape Context")
+  , matcher_type_("ransac")
+  , detector_scale_(5.0)
+  , detector_minimum_spanning_trees_num_(2.0)
+  , detector_base_sigma_(0.2)
+  , detector_sigma_step_(1.4)
+  , detector_window_(3)
+  , descriptor_min_rho_(0.02)
+  , descriptor_max_rho_(0.5)
+  , descriptor_bin_rho_(8)
+  , descriptor_bin_phi_(30)
 {
-  const double min_rho = 0.02;
-  const double max_rho = 0.5;
-  const double bin_rho = 4;
-  const double bin_phi = 12;
-  BetaGridGenerator* gen = new BetaGridGenerator(min_rho, max_rho, bin_rho,
-                                                 bin_phi);
-  gen->setDistanceFunction(dist);
-  return gen;
-}
 
+  nh_.param("detector_type",  detector_type_,  detector_type_);
```

Comment:
The parameters should be in the private namespace. Not sure if `nh_` is private. If not, use `ros::NodeHandle nh_priv("~");` inside the constructor.